src/sql/query_graph.rs

```diff
@@ -653,20 +700,23 @@ pub fn to_query_graph(st: &SelectStatement) -> Result<QueryGraph, String> {
                     value: l.clone(),
                 }));
             }
-            FieldExpression::Col(ref c) => {
-                match c.function {
-                    None => (), // we've already dealt with this column as part of some relation
-                    Some(_) => {
-                        // add a special node representing the computed columns; if it already
-                        // exists, add another computed column to it
-                        let n = qg.relations
-                            .entry(String::from("computed_columns"))
-                            .or_insert_with(
-                                || new_node(String::from("computed_columns"), vec![], st),
-                            );
-                        n.columns.push(c.clone());
-                    }
+            FieldExpression::Arithmetic(ref a) => {
+                if let ArithmeticBase::Column(ref c) = a.left {
+                    add_computed_column(&mut qg, c);
+                }
+
+                if let ArithmeticBase::Column(ref c) = a.right {
+                    add_computed_column(&mut qg, c);
                 }
+
+                qg.columns.push(OutputColumn::Arithmetic(ArithmeticColumn {
+                    name: String::from("arithmetic"),
```

Comment:
How does this deal with aliases for arithmetic columns? Seems like it might hardcode the column name always, which would be problematic if we had more than one arithmetic column.

Reply:
Hm, that's a good point. I implemented this by looking at how it was being done for literals, which also seems to use the same name for the computed columns:
https://github.com/ekmartin/distributary/blob/85044bf28c9bb83bbe2b907bfdfaf7cf0b6d0c84/src/sql/query_graph.rs#L696-L702


Both literals and arithmetic expression names are handled here though:
https://github.com/ekmartin/distributary/blob/85044bf28c9bb83bbe2b907bfdfaf7cf0b6d0c84/src/sql/mir.rs#L737-L763

Where column objects are created with only the name and the table. It doesn't seem to look at aliases at all for literals though, which I guess might mean that aliases doesn't work for literals either?

I added a test for multiple arithmetic expressions in one query [here](https://github.com/mit-pdos/distributary/pull/30/commits/28a9dadb8743cc9d18e19b7140ac83a8fdee5b39) - I'll see how it behaves with aliases as well.

Reply:
Yeah, you're right that aliases for literals probably don't work.

Maybe it's best to add alias support to both `ArithmeticColumn` and `LiteralColumn` for now (either within this PR or separately, I don't mind).